community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/TemporalIndexCache.java

```diff
@@ -151,7 +151,7 @@ T date() throws E
         if ( date == null )
         {
             date = factory.newDate();
```

Comment:
So these factories can return `null` sometimes? Are they generally costly to call, if so perhaps a flag for knowing when they have been called could be in order

Reply:
The only one which returns `null` is the `TemporalIndexReader`, which is not allowed to create new accessors (and therefore indexes) for temporal types which do not have any indexes values. In this case calling the factory is super-cheap: essentially only a switch and a null-check, so we saw no particular reason to be clever there. Will still need to check on every query though, because a new temporal value might have been added since the last query, causing the index to now exist.

We did think about refactoring here for nicer type-safety around `uncheckedSelect`, which is only used from the reader, where we statically know it will not throw, but I guess that's a slightly different concern.